### PR TITLE
remove unused imports, make clicktests more reliable, fix git rev-parse with bare repositories, fix crash when directory gets deleted

### PR DESCRIPTION
**Pull request by campersau**
(also want to check timeouts in appveyor builds)

fixes https://github.com/FredrikNoren/ungit/issues/1265
fixes https://github.com/FredrikNoren/ungit/issues/1266

### CODE REVIEW

**Comment by campersau** [comment]
It looks like git version 2.25 [changed the behavior](https://github.com/git/git/commit/2d92ab32fd624349d308334befbf07adf9f179b9) of `git rev-parse --show-toplevel` inside of `bare` repositories. It now throws an error: `fatal: this operation must be run in a work tree`

That's the reason why travis with the [latest git version](https://travis-ci.org/FredrikNoren/ungit/jobs/639954606#L1103) and [all appveyor builds](https://ci.appveyor.com/project/FredrikNoren/ungit/builds/30258417/job/sqk1iyf7x01chn8c#L117) are failing. Appveyor [updated](https://github.com/appveyor/website/commit/4e0dd91bc6d6bf26d14e3c6f3cc3c6eb5ca6126f) the git version a [few days ago](https://github.com/appveyor/ci/issues/3242#issuecomment-575083510).

File: source/git-api.js [hunk 1]
```diff
@@ -556,7 +560,7 @@ exports.registerApi = (env) => {
       .then((baseRepoPath) => {
         return { path: path.resolve(baseRepoPath.trim()) };
       }).catch((e) => {
-        if (e.errorCode === 'not-a-repository') {
+        if (e.errorCode === 'not-a-repository' || e.errorCode === 'must-be-in-working-tree') {
           return {};
```
Comment (Hirse) [inline source/git-api.js]: Maybe add a comment why this error can be ignored.

File: source/git-api.js [hunk 2]
```diff
@@ -716,13 +716,11 @@ exports.registerApi = (env) => {
...
       jsonResultOrFailProm(res, gitPromise(req.body.command, req.body.repo))
     });
-    app.post(`${exports.pathPrefix}/testing/cleanup`, ensureAuthenticated, (req, res) => {
-      //winston.info('Cleaned up: ' + JSON.stringify(cleaned));
-      res.json({ result: temp.cleanup() });
-    });
-    app.post(`${exports.pathPrefix}/testing/shutdown`, ensureAuthenticated, (req, res) => {
-      res.json({});
-      process.exit();
+    app.post(`${exports.pathPrefix}/testing/cleanup`, (req, res) => {
```
[the remark this answers is not in the document]
Comment (Hirse) [reply source/git-api.js]: Are these endpoints still available when running ungit outside of tests?
If it doesn't require authentication, should it be hidden behind a `TEST` env variable for example?

[the remark this answers is not in the document]
Comment (campersau) [reply source/git-api.js]: Yes, they are hidden behind the `dev` config (and only the `testing/cleanup` endpoint is affected).
https://github.com/FredrikNoren/ungit/blob/68d69bb150aa200dbf619c10adb9a6cdc4852131/source/git-api.js#L691
https://github.com/FredrikNoren/ungit/blob/68d69bb150aa200dbf619c10adb9a6cdc4852131/source/config.js#L65-L66

**Pull request merged**